### PR TITLE
Feat/official thing models

### PR DESCRIPTION
**Pull request by hadjian**
- Checking if Author and Manufacturer are equal before calling model.ParseTMID()
- Should this be a method of the ThingModel model?

### CODE REVIEW

File: internal/commands/fetch.go [hunk 1]
```diff
@@ -65,8 +66,9 @@ func FetchThingByName(fn *FetchName, remote remotes.Remote) ([]byte, error) {
 		}
 	}
 
-	// TODO: how to know if it is official?
-	tmid, err := model.ParseTMID(id, false)
+	official := internal.Prep(tocThing.Author.Name) == internal.Prep(tocThing.Manufacturer.Name)
```
Comment (alexbrdn) [inline internal/commands/fetch.go:69]: I wonder if calling `internal.Prep()` is necessary here.
As a side-note, "Prep" should really be renamed. Note that push [uses](https://github.com/web-of-things-open-source/tm-catalog-cli/blob/ddfa9c111fc6153780897bc83fe99bbfdca419b8/internal/commands/validate/validate.go#L72) `sanitize.BaseName()`

File: internal/toc/toc.go [hunk 2]
```diff
@@ -106,20 +107,21 @@ func saveToc(rootPath string, tocBytes []byte) error {
 }
 
 func insert(table model.Toc, ctm model.CatalogThingModel) error {
-	// TODO: if author and manufacturer are equal, it is offical, right?
-	tmid, err := model.ParseTMID(ctm.ID, false)
+	official := internal.Prep(ctm.Manufacturer.Name) == internal.Prep(ctm.Author.Name)
```
Comment (alexbrdn) [inline internal/toc/toc.go:110]: Wouild it make sense to consolidate these checks in one place?

Comment (hadjian) [reply internal/toc/toc.go:110]: Yes, which place do you propose? I suggested the ThingModel struct? Something like `func (tm *ThingModel) isOfficial()`?

Comment (alexbrdn) [reply internal/toc/toc.go:110]: I don't think that'd work for `TocThing`

File: internal/commands/fetch.go (hunk 1 above)
Comment (hadjian) [reply internal/commands/fetch.go:69]: On my TODO. I don't think `sanitize.BaseName()` serves the same purpose, though. `Prep()` trims and lower cases strings to make them comparable, e.g. it is used when filtering.

On the naming: util.ToTrimmedLower(s string) string?

File: internal/toc/toc.go (hunk 2 above)
Comment (hadjian) [reply internal/toc/toc.go:110]: Correct, I forgot that I removed the struct embedding. The `TocThing` relies on the `ThingModel`. The latter is used for parsing the fields from the TM, the latter is just a restructuring of the fields. How about we put the `isOfficial()` method with the `ThingModel` and save it's outcome into a plain field in the `TocThing`? Not perfect, yet, but maybe better than code duplication?

Comment (hadjian) [reply internal/toc/toc.go:110]: I will make a note and merge for now.

**Pull request merged**